File: packages/android/src/main/java/com/kozmos/components/Table/Table.figma.kt
package com.kozmos.components.table

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=237-2547")
class KozmosTableConnect {
    @FigmaProperty(FigmaType.Enum, "Density")
    val density: String = Figma.mapping(
        "Default" to "default",
        "Compact" to "compact"
    )

    @FigmaProperty(FigmaType.Text, "Header 1 Text")
    val header1: String = "Name"

    @FigmaProperty(FigmaType.Text, "Header 2 Text")
    val header2: String = "Status"

    @FigmaProperty(FigmaType.Text, "Header 3 Text")
    val header3: String = "Updated"

    @FigmaProperty(FigmaType.Text, "Row 1 Cell 1 Text")
    val row1Cell1: String = "Beacon sync"

    @FigmaProperty(FigmaType.Text, "Row 1 Cell 2 Text")
    val row1Cell2: String = "Ready"

    @FigmaProperty(FigmaType.Text, "Row 1 Cell 3 Text")
    val row1Cell3: String = "Today"

    @FigmaProperty(FigmaType.Text, "Row 2 Cell 1 Text")
    val row2Cell1: String = "Indoor map"

    @FigmaProperty(FigmaType.Text, "Row 2 Cell 2 Text")
    val row2Cell2: String = "Draft"

    @FigmaProperty(FigmaType.Text, "Row 2 Cell 3 Text")
    val row2Cell3: String = "Yesterday"

    @FigmaProperty(FigmaType.Text, "Row 3 Cell 1 Text")
    val row3Cell1: String = "Analytics"

    @FigmaProperty(FigmaType.Text, "Row 3 Cell 2 Text")
    val row3Cell2: String = "Queued"

    @FigmaProperty(FigmaType.Text, "Row 3 Cell 3 Text")
    val row3Cell3: String = "May 21"

    @Composable
    fun ComponentExample() {
        KozmosTable(
            header = {
                KozmosTableRow {
                    KozmosTableHeaderCell(text = header1)
                    KozmosTableHeaderCell(text = header2)
                    KozmosTableHeaderCell(text = header3)
                }
            }
        ) {
            KozmosTableRow {
                KozmosTableCell(text = row1Cell1)
                KozmosTableCell(text = row1Cell2)
                KozmosTableCell(text = row1Cell3)
            }
            KozmosTableRow {
                KozmosTableCell(text = row2Cell1)
                KozmosTableCell(text = row2Cell2)
                KozmosTableCell(text = row2Cell3)
            }
            KozmosTableRow {
                KozmosTableCell(text = row3Cell1)
                KozmosTableCell(text = row3Cell2)
                KozmosTableCell(text = row3Cell3)
            }
        }
    }
}

File: packages/ios/Sources/Components/Table/Table.figma.swift
import SwiftUI
import Figma

struct KozmosTableConnectRow: Identifiable {
    let id: String
    let first: String
    let second: String
    let third: String
}

struct KozmosTableConnect: FigmaConnect {
    let component = KozmosTable<[KozmosTableConnectRow], AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=237-2547"

    @FigmaEnum(
        "Density",
        mapping: [
            "Default": "default",
            "Compact": "compact"
        ]
    )
    var density: String = "default"

    @FigmaString("Header 1 Text")
    var header1: String = "Name"

    @FigmaString("Header 2 Text")
    var header2: String = "Status"

    @FigmaString("Header 3 Text")
    var header3: String = "Updated"

    @FigmaString("Row 1 Cell 1 Text")
    var row1Cell1: String = "Beacon sync"

    @FigmaString("Row 1 Cell 2 Text")
    var row1Cell2: String = "Ready"

    @FigmaString("Row 1 Cell 3 Text")
    var row1Cell3: String = "Today"

    @FigmaString("Row 2 Cell 1 Text")
    var row2Cell1: String = "Indoor map"

    @FigmaString("Row 2 Cell 2 Text")
    var row2Cell2: String = "Draft"

    @FigmaString("Row 2 Cell 3 Text")
    var row2Cell3: String = "Yesterday"

    @FigmaString("Row 3 Cell 1 Text")
    var row3Cell1: String = "Analytics"

    @FigmaString("Row 3 Cell 2 Text")
    var row3Cell2: String = "Queued"

    @FigmaString("Row 3 Cell 3 Text")
    var row3Cell3: String = "May 21"

    var body: some View {
        VStack(spacing: KozmosDimensions.primitivesLayoutSpacing0) {
            KozmosTableRow {
                KozmosTableCell(self.header1)
                KozmosTableCell(self.header2)
                KozmosTableCell(self.header3)
            }

            KozmosTable(rows) { row in
                AnyView(
                    KozmosTableRow {
                        KozmosTableCell(row.first)
                        KozmosTableCell(row.second)
                        KozmosTableCell(row.third)
                    }
                )
            }
        }
    }

    private var rows: [KozmosTableConnectRow] {
        [
            KozmosTableConnectRow(
                id: "row-1",
                first: self.row1Cell1,
                second: self.row1Cell2,
                third: self.row1Cell3
            ),
            KozmosTableConnectRow(
                id: "row-2",
                first: self.row2Cell1,
                second: self.row2Cell2,
                third: self.row2Cell3
            ),
            KozmosTableConnectRow(
                id: "row-3",
                first: self.row3Cell1,
                second: self.row3Cell2,
                third: self.row3Cell3
            )
        ]
    }
}
